### PR TITLE
grammar and typos

### PR DESCRIPTION
**Pull request by mchoplin**

### CODE REVIEW

File: chapter_01_domain_model.asciidoc [hunk 1]
```diff
@@ -2,7 +2,7 @@
 == Domain modelling
 
 In the prologue, we used the term "business logic layer" to describe the
-central layer of a three-layered architecture. For the rest of the book we're
+central layer of a three-layered architecture. For the rest of the book, we're
```
Comment (hjwp) [inline chapter_01_domain_model.asciidoc]: not sure about this one

File: chapter_02_repository.asciidoc [hunk 2]
```diff
@@ -86,7 +86,7 @@ approach to structuring an system that has a UI, some logic, and a database:
 
 Django's Model-View-Template structure is closely related, as is
 Model-View-Controller (MVC). In any case, the aim is to keep the layers
-separate (which is a good thing), and to have each layer depend only on the one
+separate (which is a good thing), and to have each layer depends only on the one
```
Comment (hjwp) [inline chapter_02_repository.asciidoc]: this one is defintely wrong. i'm 99% sure.

File: chapter_02_repository.asciidoc [hunk 3]
```diff
@@ -489,7 +489,7 @@ trying to ask: "what do we get for this?  And what does it cost us?".
 
 Usually at the very least we'll be introducing an extra layer of abstraction,
 and although we may hope it will be reducing complexity overall, it does add
-complexity locally, and it has a cost in terms raw numbers of moving parts and
+complexity locally, and it has a cost in terms of moving parts and
```
Comment (hjwp) [inline chapter_02_repository.asciidoc]: nice

Comment (hjwp) [reply chapter_02_repository.asciidoc]: or maybe just lose the word "raw"

File: chapter_02_repository.asciidoc [hunk 4]
```diff
@@ -538,7 +538,7 @@ TODO: not sure if this diagram is helping.
 ====
 
 
-As always we start with a test. This would probably be classified as an
+As always, we start with a test. This would probably be classified as an
```
Comment (hjwp) [inline chapter_02_repository.asciidoc]: do you have a rule you're following when adding these commas?

File: chapter_02_repository.asciidoc [hunk 5]
```diff
@@ -627,12 +627,14 @@ def test_repository_can_retrieve_a_batch_with_allocations(session):
...
-// Why the underscore in _allocations here?
+// Why the underscore in _allocations here? It was already private in the definition
+// of the Batch class in chapter 1. Maybe for consistency we want to make them all
+// private ('_') and explain that we want to access them through properties for better control?
 
 Whether or not you painstakingly write tests for every model is a judgement
 call.  Once you have one class tested for create/modify/save, you might be
 happy to go on and do the others with a minimal roundtrip test, or even nothing
-at all, if they all follow a similar pattern.  In our case, the ORM config
+at all, if they all follow a similar pattern. In our case, the ORM config
```
Comment (hjwp) [inline chapter_02_repository.asciidoc]: ok look, if I want to use two spaces after a full stop will you let me please? I mean I really don't want to dissuade you from writing these comments which I really, really appreciate, but these spaces don't even render in the published text! two spaces after a full stop is in my muscle memory...

File: chapter_02_repository.asciidoc [hunk 6]
```diff
@@ -724,6 +726,7 @@ class FakeRepository(AbstractRepository):
 ====
 
 Because we are a simple wrapper around a `set`, our methods are all one-liners.
+// this sentence looks weird ^ , is the 1st "are" the correct word?
```
Comment (hjwp) [inline chapter_02_repository.asciidoc]: probably not. thanks!

**Comment by matthieuchoplin** [comment]
Ha ha! Thanks for your comments Harry :)

File: chapter_02_repository.asciidoc (hunk 4 above)
Comment (matthieuchoplin) [reply chapter_02_repository.asciidoc]: I am not too sure, I think the part of the sentence that holds the important bit -what we actually want to say- should be identified by some punctuation (the comma here) that helps us focus on the content, as opposed to the other particules, that are there to pace the sentence maybe?

Comment (matthieuchoplin) [reply chapter_02_repository.asciidoc]: Oh Google helped me there. So the rule is "Use a comma after introductory adverbs" according to this: http://www.businessinsider.com/a-guide-to-proper-comma-use-2013-9

File: chapter_02_repository.asciidoc (hunk 5 above)
Comment (matthieuchoplin) [reply chapter_02_repository.asciidoc]: Okay, I will try to focus more on the important bits 🤘

File: chapter_02_repository.asciidoc [hunk 7]
```diff
@@ -193,7 +193,7 @@ class Allocation(Base):
 You don't need to understand SQLAlchemy to see that our pristine model is now
 full of dependencies on the ORM, and is starting to look ugly as hell besides.
 Can we really say this model is ignorant of the database? How can it be
-separate from storage concerns when our model properties are directly coupled
+separated from storage concerns when our model properties are directly coupled
```
Comment (LuRsT) [inline chapter_02_repository.asciidoc]: Not sure about this one

File: chapter_02_repository.asciidoc (hunk 3 above)
Comment (LuRsT) [reply chapter_02_repository.asciidoc]: Keeping the numbers would look wierd, I think the edit as is, is good

File: chapter_02_repository.asciidoc (hunk 4 above)
Comment (LuRsT) [reply chapter_02_repository.asciidoc]: It does look better with the comma :+1:

File: chapter_02_repository.asciidoc [hunk 8]
```diff
@@ -627,12 +627,14 @@ def test_repository_can_retrieve_a_batch_with_allocations(session):
 //TODO (DS): Picking a descriptive SKU (e.g. 'comfy-sofa') would make this a
 //bit more fun to read.
 // Worth explaining why we have to do a follow up query to get the id inserted?j
-// Why the underscore in _allocations here?
+// Why the underscore in _allocations here? It was already private in the definition
+// of the Batch class in chapter 1. Maybe for consistency we want to make them all
```
Comment (LuRsT) [inline chapter_02_repository.asciidoc]: I'd drop the "Maybe"

File: chapter_01_domain_model.asciidoc (hunk 1 above)
Comment (LuRsT) [reply chapter_01_domain_model.asciidoc]: I don't think the comma should be there :thinking:

File: chapter_02_repository.asciidoc (hunk 4 above)
Comment (hjwp) [reply chapter_02_repository.asciidoc]: the thing is, O'Reilly will have their own style guide and conventions for this sort of thing, so there's a 50-50 chance they're just going to change it back.  When the book goes through the production process a copy-editor goes through the whole thing with a fine-toothed comb, picking out little things like this...

File: chapter_02_repository.asciidoc (hunk 7 above)
Comment (hjwp) [reply chapter_02_repository.asciidoc]: both are fine tbh.

File: chapter_02_repository.asciidoc (hunk 8 above)
Comment (hjwp) [reply chapter_02_repository.asciidoc]: ok now you're making suggestions for the grammar in _comments_

Comment (hjwp) [reply chapter_02_repository.asciidoc]: not even comments i wrote!  that's a comment from a tech reviewer :rofl:

**Pull request merged**